compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -556,70 +556,86 @@ class ComposeScene internal constructor(
             PointerEventType.Scroll -> processScroll(event)
         }
 
-        val isAnyPointerDown = event.pointers.fastAny { it.down }
-        if (!isAnyPointerDown) {
+        if (!event.isAnyPointerDown) {
             pressOwner = null
         }
     }
 
     private fun processPress(event: PointerInputEvent) {
-        val owner = reversedOwners {
-            for (owner in it) {
-                if (owner.isHovered(event)) {
-                    // Stop once the position of in bounds of the owner
-                    return@reversedOwners owner
-                }
-                owner.onClickOutside?.invoke()
-                if (owner == focusedOwner) {
-                    // Stop if it's in focus, do not pass the event to hovered owner
-                    return@processPress
-                }
+        forEachOwnerReversed { owner ->
+            if (owner.isHovered(event)) {
+                // Stop once the position of in bounds of the owner
+                owner.processPointerInput(event)
+                pressOwner = owner
+                return@processPress
+            }
+            owner.onClickOutside?.invoke()
+            if (owner == focusedOwner) {
+                // Stop if it's in focus, do not pass the event to hovered owner
+                return@processPress
             }
-            return@reversedOwners null
         }
-        owner?.processPointerInput(event)
-        pressOwner = owner
     }
 
     private fun processRelease(event: PointerInputEvent) {
-        val owner = pressOwner ?: hoveredOwner(event)
-        if (focusedOwner.isAbove(owner)) {
-            return
+        pressOwner
+            .takeIf { !focusedOwner.isAbove(it) }
+            ?.processPointerInput(event)
+        if (!event.isAnyPointerDown) {
+            // Changing hover during Move event can be blocked by sticking to pressOwner
+            hoveredOwner(event)
+                .takeIf { !focusedOwner.isAbove(it) }
+                ?.let { processHover(event, it) }
```

Comment:
Can you add a big comment explaining the all the logic here? It's quite complicated, and if you don't know what it wants to do, it's impossible to say whether it does that or there's a bug.

Reply:
Actually I found one more case that should be covered here. I'll update in later today